apps/api/src/http/internal/rest/webhooks/stripe/mod.rs

```diff
@@ -0,0 +1,143 @@
+use std::sync::Arc;
+
+use axum::extract::State;
+use axum::http::{HeaderMap, StatusCode};
+use mongodb::options::UpdateOptions;
+use shared::database::queries::{filter, update};
+use shared::database::webhook_event::WebhookEvent;
+
+use crate::global::Global;
+use crate::http::error::ApiError;
+use crate::sub_refresh_job;
+use crate::transactions::{with_transaction, TransactionError};
+
+mod charge;
+mod checkout_session;
+mod customer;
+mod invoice;
+mod subscription;
+
+pub async fn handle(State(global): State<Arc<Global>>, headers: HeaderMap, payload: String) -> Result<StatusCode, ApiError> {
+	let sig = headers
+		.get("stripe-signature")
+		.and_then(|v| v.to_str().ok())
+		.ok_or(ApiError::BAD_REQUEST)?;
+
+	let event = stripe::Webhook::construct_event(&payload, sig, &global.config.api.stripe.webhook_secret).map_err(|e| {
+		tracing::error!(error = %e, "failed to parse webhook");
+		ApiError::BAD_REQUEST
+	})?;
+
+	// TODO: verify request is coming from stripe ip
+	// https://docs.stripe.com/ips#webhook-notifications
```

Comment:
Is this required if it matches the signature?

Reply:
No not really, but Stripe recommends doing it